src/reducers/posts.js

```diff
@@ -0,0 +1,40 @@
+import { actionTypes } from '../common/constants/actionTypes';
+
+const initialState = {
+  posts: [],
+  isRequestPending: null
+};
+
+const posts = (state = initialState, action) => {
+  switch (action.type) {
+    case actionTypes.SAVE_POSTS:
+      return { ...state, posts: action.payload };
+
+    case actionTypes.SAVE_POST:
+      return { ...state, posts: [...state.posts, action.payload] };
+
+    case actionTypes.REMOVE_POST:
+      return {
+        ...state,
+        posts: state.posts.filter(post => post.id != action.payload)
+      };
+
+    case actionTypes.EDIT_POST:
+      return {
+        ...state,
+        posts: state.posts.map(
+          post => (post.id === action.payload.id ? action.payload : post)
+        )
+      };
+
+    case actionTypes.IS_REQUEST_PENDING:
```

Comment:
It's more common practice is to use 3 action types for every async request. For instance, for saving post:
- SAVE_POST_REQUEST
- SAVE_POST_SUCCESS
- SAVE_POST_FAILURE

Usually _REQUEST (and _FAILURE) action types for different async request within particular part of state have the same code (just set `isRequestPending` and `error` properties in state) so you can place it in the same `case` statement.

So please, refactor your action types and reducer (eliminate `IS_REQUEST_PENDING` action type, add appropriate action types for _REQUEST and _FAILURE, add `error` property)